### PR TITLE
fix Dockerfile name not recognized issue

### LINKED ISSUE #305: appbuilder should output the progress while building the container image
The current implementation prints nothing while building the image with `docker build`, leaving the user in dark and wonder whether the tool is hanging or not. This is bad user experience. It's also less helpful with debugging pipeline failures.

### PR DESCRIPTION
**Pull request by praenubilus**
Update the default build behavior. Only Dockerfile name will be accepted as default argument. 
Fix #305 
Add Sample Usage in help info

### CODE REVIEW

File: scripts/appbuilder [hunk 1]
```diff
@@ -8,12 +8,10 @@
 function print_usage {
     echo "$(basename $0): a useful shell script for converting Docker images into Mystikos cpio archives"
     echo
-    echo "Usage: $(basename $0) [-o file] [-d [file]] [[-i] image] [-e extras] [-v]"
+    echo "Usage: $(basename $0) [-o file] [-i image] [-e extras] [-v] [[-d] file]"
```
Comment (jxyang) [inline scripts/appbuilder:11]: Some people don't like to be forced to place an argument at a particular location.

File: scripts/appbuilder [hunk 2]
```diff
@@ -8,12 +8,10 @@
...
     echo
     echo "  -d file     Build docker image from the named file"
-    echo "              Defaults to './Dockerfile' if no option specified"
```
Comment (jxyang) [inline scripts/appbuilder:14]: Why remove it?

File: scripts/appbuilder [hunk 3]
```diff
@@ -8,12 +8,10 @@
...
     echo "  -i image    Specify an existing Docker image name to build from"
-    echo "              If neither -d nor -i are specified, will look for Dockerfile in current dir"
```
Comment (jxyang) [inline scripts/appbuilder:16]: Why remove it?

File: scripts/appbuilder [hunk 4]
```diff
@@ -58,9 +56,16 @@ done
 
 ### defaults
 
-# if no options specified, assume first argument is an image name
-shift $((OPTIND-1))
-IMAGE_NAME=${IMAGE_NAME:-$@}
+if [ $# -lt 1 ]; then
+        echo "Dockerfile or ImageName has not been specified"
```
Comment (jxyang) [inline scripts/appbuilder:60]: Does this mean we can't run `appbuilder` byitself anymore?

File: scripts/appbuilder [hunk 5]
```diff
@@ -8,12 +8,10 @@
...
-    echo "Usage: $(basename $0) [-o file] [-d [file]] [[-i] image] [-e extras] [-v]"
+    echo "Usage: $(basename $0) [-o file] [-i image] [-e extras] [-v] [[-d] file]"
     echo
     echo "  -d file     Build docker image from the named file"
-    echo "              Defaults to './Dockerfile' if no option specified"
     echo "  -i image    Specify an existing Docker image name to build from"
-    echo "              If neither -d nor -i are specified, will look for Dockerfile in current dir"
     echo "  -f format   Specify the output format. Options: dir, cpio. Default: 'dir'"
     echo "  -e extras   Specify the extra options while building the container image"
     echo "  -o      Specify the name of the output file or directory"
```
Comment (jxyang) [inline scripts/appbuilder:17]: Please provide a few example usages here.

File: scripts/appbuilder (hunk 1 above)
Comment (praenubilus) [reply scripts/appbuilder:11]: the location matters only for the default situation(no other options like -o -i )

File: scripts/appbuilder (hunk 2 above)
Comment (praenubilus) [reply scripts/appbuilder:14]: As we discussed, a docker file name is mandatory for default situation. 
```bash
appbulder <dockerfile-name>
```

File: scripts/appbuilder (hunk 3 above)
Comment (praenubilus) [reply scripts/appbuilder:16]: see comments above

File: scripts/appbuilder (hunk 4 above)
Comment (praenubilus) [reply scripts/appbuilder:60]: Yes. And this situation(run appbuilder by itself) does not happen to any tests or samples in current repo.

File: scripts/appbuilder (hunk 5 above)
Comment (praenubilus) [reply scripts/appbuilder:17]: That depends on the definition for the appbuilder args. After the definition(sequence, default option) has been confirmed, I will add examples.

**Comment by jxyang** [comment]
LGTM. Please add sample commands.

Do you want to address #305 in this PR?

**Comment by praenubilus** [comment]
> LGTM. Please add sample commands.
> 
> Do you want to address #305 in this PR?

Sure. Let me include #305 fix in next commit.

**Review by jxyang — APPROVED** [review]

**Comment by jxyang** [comment]
The pipeline failure is most likely caused by this line: https://github.com/deislabs/mystikos/blob/e9db8133479baef87c75fe383400fdf66c11d643/tests/libcxx/Makefile#L31

Can you remove `$(MAKE) clean` and try again?

**Comment by praenubilus** [comment]
> The pipeline failure is most likely caused by this line:
> 
> https://github.com/deislabs/mystikos/blob/e9db8133479baef87c75fe383400fdf66c11d643/tests/libcxx/Makefile#L31
> 
> Can you remove `$(MAKE) clean` and try again?

The root cause is in this line, it will break the new enforcement. The `-i` option is needed. I have fixed it with a new commit.

https://github.com/deislabs/mystikos/blob/e9db8133479baef87c75fe383400fdf66c11d643/tests/libcxx/Makefile#L49

**Pull request merged**